src/components/selectionBoxLayer.ts

```diff
@@ -111,6 +131,101 @@ export module Components {
     public fixedHeight() {
       return true;
     }
+
+    /**
+     * Gets the x scale for this SelectionBoxLayer.
+     */
+    public xScale<D extends number | { valueOf(): number }>(): QuantitativeScale<D>;
+    /**
+     * Sets the x scale for this SelectionBoxLayer.
+     * 
+     * @returns {SelectionBoxLayer} The calling SelectionBoxLayer.
+     */
+    public xScale<D extends number | { valueOf(): number }>(xScale: QuantitativeScale<D>): SelectionBoxLayer;
+    public xScale<D extends number | { valueOf(): number }>(xScale?: QuantitativeScale<D>): any {
+      if (xScale == null) {
+        return this._xScale;
+      }
+      if (this._xScale != null) {
+        this._xScale.offUpdate(this._adjustBoundsCallback);
+      }
+      this._xScale = xScale;
+      this._xScale.onUpdate(this._adjustBoundsCallback);
+      this._bindBoxDataValues();
+      return this;
+    }
+
+    /**
+     * Gets the y scale for this SelectionBoxLayer.
+     */
+    public yScale<D extends number | { valueOf(): number }>(): QuantitativeScale<D>;
+    /**
+     * Sets the y scale for this SelectionBoxLayer.
+     * 
+     * @returns {SelectionBoxLayer} The calling SelectionBoxLayer.
+     */
+    public yScale<D extends number | { valueOf(): number }>(yScale: QuantitativeScale<D>): SelectionBoxLayer;
+    public yScale<D extends number | { valueOf(): number }>(yScale?: QuantitativeScale<D>): any {
+      if (yScale == null) {
+        return this._yScale;
+      }
+      if (this._yScale != null) {
+        this._yScale.offUpdate(this._adjustBoundsCallback);
+      }
+      this._yScale = yScale;
+      this._yScale.onUpdate(this._adjustBoundsCallback);
+      this._bindBoxDataValues();
+      return this;
+    }
+
+    /**
+     * Gets the data value backing the left edge of the box.
+     *
+     * Returns null if the edge is not backed by a scale.
+     */
+    public boxLeftDataValue() {
+      return this._boxLeftDataValue;
+    }
+
+    /**
+     * Gets the data value backing the right edge of the box.
+     *
+     * Returns null if the edge is not backed by a scale.
+     */
+    public boxRightDataValue() {
+      return this._boxRightDataValue;
+    }
+
+    /**
+     * Gets the data value backing the top edge of the box.
+     *
+     * Returns null if the edge is not backed by a scale.
+     */
+    public boxTopDataValue() {
```

Comment:
Instead of having `boxTopDataValue()` and `boxBottomDataValue()`, how about having `yExtent()`?

Also, what gets returned if `xScale()` hasn't been set?